src/main.c

```diff
@@ -49,6 +52,61 @@ static void softReset_button_callback(struct device *port, struct gpio_callback
 	soft_reboot();
 }
 
+
+void cdc_acm_descriptor_callback (cdc_acm_cfg_t *cfg)
+{
+	customer_data_t *otp_data = (customer_data_t *)CUSTOMER_DATA_ADDRESS;
+
+	memset(cfg->vendor_string, '\0', 32);
```

Comment:
@SidLeung vendor string will always contain a valid string by default.

cfg->vendor_string is not a location in the OTP flash, it is a config structure used by  the cdc_acm driver; this memset call is just overwriting the initialized value which can be seen here https://github.com/01org/zephyr/blob/master/subsys/usb/class/cdc_acm.c#L243

So really, the only reason Dino needs this memset call is to ensure his string is null terminated. So in that case, since the length of the string is known, we should just explicitly null-terminate the string here.